lib/MusicBrainz/Server/Entity/URL/AmazonMusic.pm

```diff
@@ -0,0 +1,39 @@
+package MusicBrainz::Server::Entity::URL::AmazonMusic;
+
+use Moose;
+
+extends 'MusicBrainz::Server::Entity::URL';
+with 'MusicBrainz::Server::Entity::URL::Sidebar';
+
+use DBDefs;
+
+sub pretty_name
+{
+    my $self = shift;
+
+    if ($self->url =~ m{^(?:https?:)?//music.(.*?\.)([a-z]+)(?:\:[0-9]+)?/.*/(?:[0-9B][0-9A-Z]{9})(?:[^0-9A-Z]|$)}i) {
```

Comment:
Same re: using a pattern closer to the one from the URLCleanup file. But I don't think it needs to match anything after the TLD here, since that's all you need for the country. And regarding the country, would this convert TLDs like `.co.uk` and `.co.jp` correctly?

Reply:
I changed this to use the other pattern and changed how it converts the TLD to be safe.